### PR TITLE
APP-4672 Improve input validator

### PR DESCRIPTION
**Pull request by symphony-annaLlorens**
### Description
- Fix required validator to Dropdown
- Increase test cases to Input validators

### Before
Screenshot 2021-11-09 at 14 14 33

### After fix

Screenshot 2021-11-09 at 14 15 16

[Jira-ticket](https://perzoinc.atlassian.net/browse/APP-4672)

### CODE REVIEW

File: src/core/validators/validators.ts [hunk 1]
```diff
@@ -119,8 +121,8 @@ const Email: ValidatorFn = (value) => {
   return Promise.resolve(null);
 };
 
-const isEmpty = (value: string | any[]) => {
-  return !value?.length || (typeof value==='string' && value?.trim?.() === '');
+const isEmptyValue = (value: Value) => {
+  return !value || (typeof value==='string' && value?.trim?.() === '') || Object.keys(value).length === 0;
```
Comment (symphony-adnane) [inline src/core/validators/validators.ts:125]: This won't work for an empty array

Comment (symphony-adnane) [reply src/core/validators/validators.ts:125]: could combine two first as : !value?.trim?()

File: src/core/validators/validators.ts [hunk 2]
```diff
@@ -57,7 +59,7 @@ const MinLength = (minlength: number): ValidatorFn => {
  */
 const MaxLength = (maxLength: number): ValidatorFn => {
   return (value) => {
-    if (!value|| maxLength >= value.length) {
+    if (!value|| maxLength >= value.length || Object.getPrototypeOf(value) === Object.prototype) {
```
Comment (symphony-adnane) [inline src/core/validators/validators.ts:62]: you could use lodash isEmpty for all these checks imo

File: src/core/validators/validators.ts [hunk 3]
```diff
@@ -31,7 +33,7 @@ const EmptyString: ValidatorFn = (value) => {
   console.warn(
     'Calling a deprecated validator (EmptyString), please use the Required validator instead'
   );
-  if (isEmpty(value)) {
+  if (isEmptyValue(value)) {
```
Comment (symphony-adnane) [inline src/core/validators/validators.ts:36]: https://lodash.com/docs/4.17.15#isEmpty

File: src/core/validators/validators.ts (hunk 2 above)
Comment (symphony-annaLlorens) [reply src/core/validators/validators.ts:62]: Here, what I'm checking is that is an object strict. If I use the lodash ._isObject() it does not differentiate [] from {}

File: src/core/validators/validators.ts (hunk 3 above)
Comment (symphony-annaLlorens) [reply src/core/validators/validators.ts:37]: isEmptyValue() is also checking that the string doesn't contain spaces. 
The isEmpty() from lodash is not checking it.

Comment (symphony-adnane) [reply src/core/validators/validators.ts:37]: you may wanna trim it if possible before passing it ...

File: spec/core/validators/validators.spec.tsx [hunk 4]
```diff
@@ -1,49 +1,82 @@
...
     expect(await Validators.MaxLength(3)('abcd')).toEqual({ maxLength: true });
+    expect(await Validators.MaxLength(3)(stringWithValue)).toEqual(null);
+    expect(await Validators.MaxLength(3)(null)).toEqual(null);
+    expect(await Validators.MaxLength(3)(undefined)).toEqual(null);
+    expect(await Validators.MaxLength(3)(emptyString)).toEqual(null);
+    expect(await Validators.MaxLength(3)(objectWithValue)).toEqual(null);
+    expect(await Validators.MaxLength(3)(emptyObject)).toEqual(null);
+    expect(await Validators.MaxLength(3)(emptyArray)).toEqual(null);
+    expect(await Validators.MaxLength(2)(arrayWithValues)).toEqual(null);
+    expect(await Validators.MaxLength(1)(arrayWithValues)).toEqual( {maxLength: true});
```
Comment (symphony-adnane) [inline spec/core/validators/validators.spec.tsx:75]: are minlength 0 and maxlength 0 something that should exist? or should we throw an exception?

**Review by symphony-adnane — APPROVED** [review]

File: spec/core/validators/validators.spec.tsx (hunk 4 above)
Comment (symphony-annaLlorens) [reply spec/core/validators/validators.spec.tsx:75]: Added an exception

**Review by AliciaSymphony — APPROVED** [review]

**Pull request merged**